core/operator.go

```diff
@@ -77,29 +98,43 @@ func (o *Operator) start() {
 		case <-o.quit:
 			return
 		case <-ticker.C:
+
 			if txs := o.txPool.Content(); len(txs) == 0 {
 				continue
+			} else {
+				fmt.Printf("%v\n", "operator txs..")
+				o.txsCh <- txs
 			}
-			fmt.Printf("%v\n", "operator txs..")
-			o.Seal()
 		}
 	}
 }
 
+func (o *Operator) AddTxs(txs types.Transactions) {
+	if len(txs) == 0 {
+		return
+	}
+	o.txsCh <- txs
+}
+
 // Stop sends a sigal to stop the operator
 func (o *Operator) Stop() {
 	close(o.quit)
 }
 
 // Seal get txs from txpool and construct block, then seal the block
-func (o *Operator) Seal() error {
-	block := o.constructBlock()
+func (o *Operator) Seal(txs types.Transactions) error {
+	block := o.constructBlock(txs)
 	newDbTx := o.chain.db.BeginTx()
 	if !newDbTx {
 		return errors.New("database race detected, there should be no tx pending when plasma operator commit a block")
 	}
 	// append block to chain and update chain head
 	if err := o.chain.WriteBlock(block); err != nil {
+		err := o.chain.db.RollbackTx()
```

Comment:
this `err` will overwrite the `err` in the outer `if` block.

I think this rollback error can be ignored, and log an error message.